runtime/oti/j9port.h

```diff
@@ -848,6 +848,7 @@ typedef struct J9CacheInfoQuery {
 #define J9PORT_SIG_VALUE_STRING OMRPORT_SIG_VALUE_STRING
 #define J9PORT_SIG_VALUE_ADDRESS OMRPORT_SIG_VALUE_ADDRESS
 #define J9PORT_SIG_FLAG_SIGQUIT OMRPORT_SIG_FLAG_SIGQUIT
+#define J9PORT_SIG_FLAG_SIGUSR2 OMRPORT_SIG_FLAG_SIGUSR2
```

Comment:
Perhaps we should just use `OMRPORT_SIG_FLAG_SIGUSR2` rather than adding an alias?

Reply:
That's fine with me, but then it would look different than all the others related to signals; is that okay? I haven't reviewed why there's this extra layer of define indirection.

Reply:
Signal values changed significantly in https://github.com/eclipse/omr/pull/3921 (over 3 years ago), and DTFJ/DDR were not updated properly at that time. Many references to `J9PORT_SIG_FLAG_SIGXXX` in Java code can be replaced by `J9PortLibrary.OMRPORT_SIG_FLAG_SIGXXX` (where `OMRPORT_SIG_FLAG_SIGXXX` is mentioned in `superset-constants.dat`. I'll have to think a bit about what we can do to (retroactively) fix the others.

Reply:
Let's say JVM X creates a core dump on Windows with the J9 generic signal number `OMRPORT_SIG_FLAG_SIGUSR2` which is `0x4000000`. Next, let's say JVM Y was built on Linux and thus has `OMRPORT_SIG_FLAG_SIGURG` set to `(OMRPORT_SIG_SMALLEST_SIGNAL_FLAG << 15) | OMRPORT_SIG_FLAG_IS_ASYNC` which is `0x80004`. If JVM Y uses `jdmpview` to load the core dump from JVM X, then it seems that checking for `J9PortLibrary.OMRPORT_SIG_FLAG_SIGUSR2` wouldn't help as that would be `0x80004`. Wouldn't the answer be that `resolveGenericSignal` needs an operating system enum parameter and then `resolveGenericSignal` would need to copy in the `OMRPORT_SIG_FLAG_SIG*` constants for each operating system and `switch` based on the incoming type?

Reply:
The constants in the package `com.ibm.j9ddr.vm29.structure` are included in the DDR blob within the core file, so the constants represent those from the VM that produced the core file, not the VM examining the core file. We just need to use them instead of hard-coded constants in `J9DDRImageProcess`, for example.

The only issues will be respect to flags that are new or not already encoded in existing core files. `CompatibilityConstants29.dat` can ensure those constants are available; zeros can be a sign the core file predates their introduction.

Reply:
I could also add `resolveGenericSignal` into the `com.ibm.dtfj.image.ImageProcess` interface so that `J9DDRImageProcess` will have it, and `NodeGPF` has an `ImageProcess _process` member that it could call it with. But `PlatformSectionParser` is for processing javacores and I don't find an obvious way to get from there into a `com.ibm.dtfj.image.ImageProcess`.

Reply:
Oh, I see NodeGPF is part of XML processing of older core dump types, so the `ImageProcess _process` is not an instance of `J9DDRImageProcess`. So then I guess the deeper question is: for NodeGPF and PlatformSectionParser, won't they never have access to J9PortLibrary since that's only available in core dumps?

Reply:
The more I think about this, the more it seems to me the solution is the following. What do you think?

1. Leave `NodeGPF` as-is: This is processing older core dump XMLs with fixed values of the signals.
2. For PlatformSectionParser, since we don't know the right OMR signal values, use the `1XHEXCPCODE    Signal_Number: 0000000B` instead of the `1XHEXCPCODE    J9Generic_Signal_Number`. If the former is not available, do not use the latter.
3. For J9DDRImageProcess, use J9PortLibrary

Reply:
That plan seems reasonable.

Reply:
Thanks, I guessed as much earlier today and submitted a new push for review